### DISCUSSION TITLE
Connection aborts when connectiong to Akamai edge

### DISCUSSION THREAD

**Discussion by Pascal7z** [cefsharp/CefSharp#3535] [General]
Hi @amaitland,

I have a situation when using cefsharp 89 (same also with 88). When connecting through a Squid proxy to an Akamai edge server, the TLS handshakes works well, but akamai then sends a connection termination. 
Doing the same with a similar chromium version, the browser starts exactly the same, and gets a connection termination. But after a timeout (and displaying an error message), it restarts the communication with Akamai, succeeds. 
Am I right to suspect the underlying libcef and chromium layers? Who should I contact? Any suggestion? 

Cheers, Pascal

**Comment by amaitland** [comment]
> Squid proxy to an Akamai edge server, 

What is Akamai Edge exactly? 

> But after a timeout (and displaying an error message), it restarts the communication with Akamai, succeeds.

What do you mean after a timeout? What timeout? What error message?

CEF by default has no error messaging or retry functionality, that's functionality that you would need to implement. It's not clear exactly what's going on here, perhaps you need to implement your own retry?

> Am I right to suspect the underlying libcef and chromium layers? Who should I contact? Any suggestion?

CEF has it's own support forum at https://magpcss.org/ceforum/index.php

**Reply by Pascal7z** [reply]
CEF Support Forum : Thanks for the hint. 

About Akamai Edge : This is a Load Balancer / Reverse Proxy that also acts as SSL/TLS Endpoint. 

The timeout message / code is : ERR_CONNECTION_CLOSED... It is displayed 3 Seconds, and then Chromium retries the connection and succeeds. 

About the retry functionality : That's what I was fearing. So there is some work to do on my side...

**Reply by amaitland** [reply]
Is the LoadError event called? 

http://cefsharp.github.io/api/89.0.x/search.html?SearchText=LoadError

If it is then you can try implemting your own retry functionality.

**Reply by Pascal7z** [reply]
That is a very good hint. I will investigate that. Thanks for the moment!
